resultsdbupdater/consumer.py

```diff
@@ -22,29 +22,25 @@ def log_msg(self, msg):
         LOGGER.debug(str(msg))
 
     def consume(self, msg):
-        resultsdb_style_topics = [
-            '/topic/VirtualTopic.eng.platformci.covscan.result',
-            '/topic/VirtualTopic.eng.platformci.rpmdiff.analysis.result',
-            '/topic/VirtualTopic.eng.platformci.rpmdiff.comparison.result'
-        ]
         if msg['topic'] == '/topic/VirtualTopic.eng.platformci.tier1.result':
             self.log_msg(msg)
             return ci_metrics_post_to_resultsdb(msg)
         elif msg['topic'] == '/topic/VirtualTopic.eng.cips':
             self.log_msg(msg)
             return cips_post_to_resultsdb(msg)
-        elif msg['topic'] in resultsdb_style_topics:
-            self.log_msg(msg)
-            return resultsdb_post_to_resultsdb(msg)
-        elif msg['topic'] == '/topic/VirtualTopic.qe.ci.jenkins':
+        else:
             # Some of the messages here can be empty strings, so only process
             # them if they are dicts to avoid tracebacks
             if isinstance(msg['body']['msg'], dict):
-                result_keys = msg['body']['msg'].get('results', {}).keys()
-                # From our understanding, we are only interested in the AMI
-                # test results in this topic
-                if result_keys and result_keys[0].startswith('dva.ami'):
+                single_result_keys = set([
+                    'data', 'outcome', 'ref_url', 'testcase'])
+                bulk_results_keys = set(['results', 'ref_url'])
+                actual_keys = set(msg['body']['msg'].keys())
+                if actual_keys in [single_result_keys, bulk_results_keys]:
```

Comment:
Would it be better as a subset check? That is, as long as the message contains *at least* the expected keys for a single Resultsdb message or a bulk Resultsdb message, try to handle it?

If I am reading this right, this version will ignore any message that includes any extra keys. Which is not great for forwards-compatibility.